src/mmw/js/src/modeling/views.js

```diff
@@ -1032,15 +1009,40 @@ var ResultsTabContentView = Marionette.LayoutView.extend({
     },
 
     onShow: function() {
-        var modelPackage = App.currentProject.get('model_package'),
+        var self = this,
+            modelPackage = App.currentProject.get('model_package'),
             resultName = this.model.get('name'),
-            ResultView = getResultView(modelPackage, resultName);
-
-        this.resultRegion.show(new ResultView({
-            model: this.model,
-            areaOfInterest: this.options.areaOfInterest,
-            scenario: this.scenario
-        }));
+            ResultView = getResultView(modelPackage, resultName),
+            tmvModel = new coreModels.TaskMessageViewModel(),
+            polling = this.model.get('polling'),
+            result = this.model.get('result'),
+            error = self.scenario.get('poll_error');
+
+        // Only show this on the initial polling. On subsequent polling, we
+        // keep showing the current results.
+        if (!result) {
```

Comment:
Hrm the multi-level nesting, while nicely efficient, is pretty hard to read. I tried some variations, but this is the best I could come up with:

```js
if (result) {
    this.resultRegion.show(new ResultView({
        model: this.model,
        areaOfInterest: this.options.areaOfInterest,
        scenario: this.scenario,
    }));
    return;
}

tmvModel.setWorking('Gathering Data');

if (error) {
    if (error.timeout) {
        tmvModel.setTimeoutError();
    } else {
        var message = error === 'NO_LAND_COVER' ?
            'Selected area of interest doesn\'t include any land cover to run the model' :
            'Error';
        tmvModel.setError(message);
    }
}

if (polling) {
    tmvModel.setWorking('Calculating Results');
}

self.resultRegion.show(new coreViews.TaskMessageView({ model: tmvModel }));
```